glustercli/cmd/root.go

```diff
@@ -149,3 +153,31 @@ func (gOpt *GlustercliOption) SetSecret() {
 	}
 	gOpt.Secret = string(data)
 }
+
+// SetEndpoints will Set the endpoints based on precedence.
+// Endpoints are taken in following order of precedence (highest to lowest):
+// --endpoints
+// GLUSTERD2_ENDPOINTS (environment variable)
+// GD2_CLIENTADDRESS (environment variable)
+// (default value)
+func (gOpt *GlustercliOption) SetEndpoints() {
+	// --endpoints
+	if len(gOpt.Endpoints) > 0 {
+		return
+	}
+
+	// GLUSTERD2_ENDPOINTS
+	if endpoint := os.Getenv("GLUSTERD2_ENDPOINTS"); endpoint != "" {
+		gOpt.Endpoints = strings.Split(endpoint, ",")
```

Comment:
Added `GLUSTERD2_ENDPOINTS` to match `GLUSTERD2_AUTH_SECRET` env variable naming convention.

I can add http, but what if https is used later? Can we change the env variable in gcs deploy to add complete endpoint?

Reply:
if we add HTTP schema to env variable, we need to make code changes to remove schema during cluster joining

Reply:
cluster joining will use different port right? name indicates it is client address.

Reply:
we need to test this changes in GCS also

Reply:
PEER_ADDRESS is set here https://github.com/gluster/gcs/blob/748762d5eab51f84398cea8817f23d912b901a6a/deploy/templates/gcs-manifests/gcs-gd2.yml.j2#L50

Reply:
The change looks fine. It'll be good if we can stick to one common prefix - `GD2` or `GLUSTERD2` or even better, explicitly specify `GLUSTERCLI` as prefix for env variables used by `glustercli`.

This can be automated as follows for all CLI options:
https://github.com/gluster/glusterd2/blob/master/glusterd2/config.go#L133-L138

@Madhu-1 
Apart from convenience, is there any other reason to derive `GLUSTERD2_ENDPOINTS` from `GD2_CLIENTADDRESS` ? We can rather simply set `GLUSTERD2_ENDPOINTS` (or `GLUSTERCLI_ENDPOINTS`) directly and explicitly.

Reply:
nothing. if we support `GLUSTERD2_ENDPOINTS` env that would be fine as well.

Reply:
`GD2_CLIENTADDRESS` is currently in use in GCS setup so added support for the same. But in CLI we already have one env variable like `GLUSTERD2_AUTH_SECRET`. To match the naming convention for general purpose use I added `GLUSTERD2_ENDPOINTS`. glusterd2 server uses some env variables with `GD2_` prefix.

+1 for automation part.

I will change all env variables in `glustercli` to use `GD2_`(instead of `GLUSTERD2_`) and change this variable as `GD2_ENDPOINTS`. With that change we need change in gcs deployment script to set `GD2_ENDPOINTS`, I will send the PR to gcs repo.